### PR TITLE
Implement Atomics.isLockFree, wait, notify

### PR DESCRIPTION
**Pull request by ksh8281**

### CODE REVIEW

File: src/builtins/BuiltinAtomics.cpp [hunk 1]
```diff
@@ -435,6 +436,179 @@ static Value builtinAtomicsXor(ExecutionState& state, Value thisValue, size_t ar
...
+    // 11. Let S be RemoveWaiters(WL, c).
+    // 12. Repeat, while S is not an empty List,
+    //     a. Let W be the first agent in S.
+    //     b. Remove W from the front of S.
+    //     c. Perform NotifyWaiter(WL, W).
+    //     d. Set n to n + 1.
+    for (n = 0; n < count; n++) {
+        WL->m_mutex.unlock();
+        WL->m_waiter.notify_one();
+        WL->m_mutex.lock();
```
Comment (clover2123) [inline src/builtins/BuiltinAtomics.cpp:589]: Is it necessary to repeatedly lock and unlock between the notify operation here?

File: src/shell/Shell.cpp [hunk 2]
```diff
@@ -540,6 +540,13 @@ static ValueRef* builtin262AgentSleep(ExecutionStateRef* state, ValueRef* thisVa
 
 class ShellPlatform : public PlatformRef {
 public:
+    bool canBlock;
+
+    ShellPlatform()
+        : canBlock(true)
```
Comment (clover2123) [inline src/shell/Shell.cpp:546]: (Minor suggestion) How about encapsulating this `canBlock` member like defining it as private member and adding getter/setter methods?

File: src/builtins/BuiltinAtomics.cpp (hunk 1 above)
Comment (ksh8281) [reply src/builtins/BuiltinAtomics.cpp:589]: unlock, lock is needed by spec. :(

File: src/shell/Shell.cpp (hunk 2 above)
Comment (ksh8281) [reply src/shell/Shell.cpp:546]: I can modify it

File: src/builtins/BuiltinAtomics.cpp (hunk 1 above)
Comment (clover2123) [reply src/builtins/BuiltinAtomics.cpp:589]: I see :)

File: src/shell/Shell.cpp (hunk 2 above)
Comment (ksh8281) [reply src/shell/Shell.cpp]: I updated it

**Review by clover2123 — APPROVED** [review]
LGTM 👍

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**